opencaesar/oml-vision
# feat/56-property-commands

## Checklist before submitting a merge request


## Description of contribution

This feature allows users to add custom commands (SPARQL queries) and run these commands in the property sheet view.

closes #56 

## Testing performed



- [x] I have added unit tests to cover additional functionality (backend only, frontend tests still need to be added)

## How to Test Expected functionality changes

1. Go to https://github.com/pogi7/kepler16b-example
2. Switch to `main` branch
3. Run `git pull`
4. From the command line run `./gradlew load`
5. Wait for that command to finish
6. Open the `Objectives` webview
7. Click the `OML Vision Properties` panel
8. Click on any row to select the row then right click the selected row
9. From the property view update the `Label` of the row to `TEST` and press Enter on your keyboard
10. You should see the table update
11. From the command line run `./gradlew save`
12. You should see the `objectives.oml` file change.  You can run `git status` to verify.

## Additional context

[If needed, you may add additional context]
